satpy/tests/reader_tests/test_ami_l1b.py

```diff
@@ -0,0 +1,308 @@
+               'sensor': 'ami',
+               'units': '1'}
+        for key, val in exp.items():
+            self.assertEqual(val, res.attrs[key])
+        self._check_orbital_parameters(res.attrs['orbital_parameters'])
+
+
+class TestAMIL1bNetCDFIRCal(TestAMIL1bNetCDFBase):
+    """Test IR specific things about the AMI reader."""
+
+    def setUp(self):
+        """Create test data for IR calibration tests."""
+        count_data = (np.arange(10).reshape((2, 5))) + 7000
+        count_data = count_data.astype(np.uint16)
+        count = xr.DataArray(
+            da.from_array(count_data, chunks='auto'),
+            dims=('y', 'x'),
+            attrs={
+                'channel_name': "IR087",
+                'detector_side': 2,
+                'number_of_total_pixels': 484000000,
+                'number_of_error_pixels': 113892451,
+                'max_pixel_value': 32768,
+                'min_pixel_value': 6,
+                'average_pixel_value': 8228.98770845248,
+                'stddev_pixel_value': 13621.130386551,
+                'number_of_total_bits_per_pixel': 16,
+                'number_of_data_quality_flag_bits_per_pixel': 2,
+                'number_of_valid_bits_per_pixel': 13,
+                'data_quality_flag_meaning':
+                    "0:good_pixel, 1:conditionally_usable_pixel, 2:out_of_scan_area_pixel, 3:error_pixel",
+                'ground_sample_distance_ew': 1.4e-05,
+                'ground_sample_distance_ns': 1.4e-05,
+            }
+        )
+        super(TestAMIL1bNetCDFIRCal, self).setUp(counts=count)
+
+    def test_ir_calibrate(self):
+        """Test IR calibration."""
+        from satpy import DatasetID
+        ds_id = DatasetID(name='IR087', wavelength=[8.415, 8.59, 8.765],
+                          calibration='brightness_temperature')
+        ds_info = {
+            'file_key': 'image_pixel_values',
+            'wavelength': [8.415, 8.59, 8.765],
+            'standard_name': 'toa_brightness_temperature',
+            'units': 'K',
+        }
+        res = self.reader.get_dataset(ds_id, ds_info)
+        expected = np.array([[238.34385135, 238.31443527, 238.28500087, 238.25554813, 238.22607701],
+                             [238.1965875, 238.16707956, 238.13755317, 238.10800829, 238.07844489]])
+        np.testing.assert_allclose(res.data.compute(), expected, equal_nan=True)
+        # make sure the attributes from the file are in the data array
+        self.assertEqual(res.attrs['standard_name'], 'toa_brightness_temperature')
+
+        # test builtin coefficients
+        self.reader.calib_mode = 'FILE'
+        res = self.reader.get_dataset(ds_id, ds_info)
+        # file coefficients are pretty close, give some wiggle room
+        np.testing.assert_allclose(res.data.compute(), expected, equal_nan=True, atol=0.04)
+        # make sure the attributes from the file are in the data array
+        self.assertEqual(res.attrs['standard_name'], 'toa_brightness_temperature')
```

Comment:
This doesn't seem to be checking if the "PYSPECTRAL" method actually calls pyspectral ?

Reply:
No, I suppose you're right. It tests what I want it to test, but doesn't actually make sure it called the right code blocks. I'm not sure the easiest way to test that. In my experience mocking and wrapping a class method is not fun. Hm I suppose I could mock pyspectral's method.

Reply:
Done.